Sources/Server/Response.swift

```diff
@@ -2,60 +2,62 @@ import Foundation
 
 public class Response {
 
-   public var queries: [String:String] = ["noun": "World"]
-   public var paths = ["/","/hello"]
-   public let status200 = "200 OK"
-   public let status404 = "404 Not Found"
-   public let status404Body = "404 Page Not Found"
-
-   public init(){}
+    let parser: Parser
+    var paths: Array<String>
+    let status200: String
+    let status404: String
+    let status404Body: String
+
+    public init(parser: Parser){
+        self.parser = parser
+        self.paths = ["/","/hello"]
+        self.status200 = "200 OK"
+        self.status404 = "404 Not Found"
+        self.status404Body = "404 Page Not Found"
+    }
 
    public func buildResponse(serverRequest: String) -> String {
-       let (requestMethod, requestPath) = parseRequest(request: serverRequest)
-       return  prepareResponse(method: requestMethod, path: requestPath)
+        let (requestMethod, requestPath) = parser.parseRequest(request: serverRequest)
+        return  prepareResponse(method: requestMethod, path: requestPath)
    }
 
    private func prepareResponse(method: String, path: String) -> String {
-       let body: String
-       let header: String
-       if (paths.contains(path)) {
-           body = prepareBody(body: "Hello \(queries["noun"]!)", method: method)
-           header = buildHeader(statusCode: status200, contentLength: body.utf8.count) 
-           queries.updateValue("World", forKey: "noun")
-       } else {
-           body = prepareBody(body: status404Body, method: method) 
-           header = buildHeader(statusCode: status404, contentLength: body.utf8.count) 
-       }      
-       return header + body
+        let body: String
+        let header: String
+        if (paths.contains(path)) {
+            body = prepareBody(body: "Hello\(buildBody())", method: method)
+            header = buildHeader(statusCode: status200, contentLength: body.utf8.count) 
+            parser.setDefaultQueries()
+        } else {
+            body = prepareBody(body: status404Body, method: method) 
+            header = buildHeader(statusCode: status404, contentLength: body.utf8.count) 
+        }      
+        return header + body
    }
 
-   private func prepareBody(body: String, method: String) -> String {
-       if (body == status404Body && method == "HEAD" || method == "HEAD") {
-           return ""    
-       } else {
-           return "<!DOCTYPE html><html><body><h1>\(body)</h1></body></html>"
-       }
+   private func buildBody() -> String {
+        var body = ""
+        for (_, value) in parser.queries {
+            if (value != ""){
+                body = body + " " + String(value)
+            }
+        }
+        return body
```

Comment:
After our earlier conversation, I was expecting you to use `joined`. :-) Anyway, here's an example of how you might avoid looping and conditional checks:

```
➜  Swift-Server git:(parse_multiple_queries) swift
Welcome to Apple Swift version 4.0.3 (swiftlang-900.0.74.1 clang-900.0.39.2). Type :help for assistance.
  1> import Foundation
  2> let names = [" Mary  ", "Jane", "     Smith"]
names: [String] = 3 values {
  [0] = " Mary  "
  [1] = "Jane"
  [2] = "     Smith"
}
  3> names.map({$0.trimmingCharacters(in: .whitespacesAndNewlines)}).joined(separator: " ")
$R0: String = "Mary Jane Smith"
  4>
```